internal/api/convert/v1alpha.go

```diff
@@ -0,0 +1,154 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"encoding/json"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha2"
+)
+
+func V1Alpha1to2(in v1alpha1.OpenTelemetryCollector) v1alpha2.OpenTelemetryCollector {
+	copy := in.DeepCopy()
+	out := v1alpha2.OpenTelemetryCollector{
+		TypeMeta:   copy.TypeMeta,
+		ObjectMeta: copy.ObjectMeta,
+	}
+
+	collectorJson, err := yaml.YAMLToJSON([]byte(in.Spec.Config))
+	if err != nil {
+		return v1alpha2.OpenTelemetryCollector{}
+	}
+
+	cfg := &v1alpha2.Config{}
+	if err := json.Unmarshal(collectorJson, cfg); err != nil {
+		return v1alpha2.OpenTelemetryCollector{}
+	}
+	out.Spec.Config = *cfg
+
+	out.Spec.OpenTelemetryCommonFields.ManagementState = v1alpha2.ManagementStateType(copy.Spec.ManagementState)
+	out.Spec.OpenTelemetryCommonFields.Resources = copy.Spec.Resources
+	out.Spec.OpenTelemetryCommonFields.NodeSelector = copy.Spec.NodeSelector
+	out.Spec.OpenTelemetryCommonFields.Args = copy.Spec.NodeSelector
+	out.Spec.OpenTelemetryCommonFields.Replicas = copy.Spec.Replicas
+
+	if copy.Spec.Autoscaler != nil {
+		metrics := make([]v1alpha2.MetricSpec, len(copy.Spec.Autoscaler.Metrics))
+		for i, m := range copy.Spec.Autoscaler.Metrics {
+			metrics[i] = v1alpha2.MetricSpec{
+				Type: m.Type,
+				Pods: m.Pods,
+			}
+		}
+		out.Spec.OpenTelemetryCommonFields.Autoscaler = &v1alpha2.AutoscalerSpec{
```

Comment:
do we need to worry about the deprecated min and max replica fields?

Reply:
I did say no. Since you are still able to set min/max in the AutoscalerSpec.

Reply:
I would prefer to keep the change as minimal as possible. We can add/remove things afterwards.